### PR TITLE
Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by RashidMenhas**
### Followed The Project requirements :+1: 

- [ ] Add the following validations

:diamond_shape_with_a_dot_inside: For the User model:
- Name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.

:diamond_shape_with_a_dot_inside: For the Post model:
- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.

:diamonds: Add unit specs for all of the models' methods and validations.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class Post < ApplicationRecord
+  validates :title, presence: true
```
Comment (bobb-Rob) [inline app/models/post.rb:2]: - [ ] Well done adding this validation, but there is a missing validation - _title must not exceed 250_

image

Please include this missing validation.

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  end
+
+  it ' Should test posts_counter to be >= 0' do
+    expect(subject.post_counter).to be >= 0
+  end
+
+  it 'Has its posts_counter value is 0 by default' do
+    expect(subject.post_counter).to eq 10
+    expect(subject).to be_valid
+  end
```
Comment (bobb-Rob) [inline spec/models/user_spec.rb:35]: - [ ] Please include unit test for this method in User model 👇🏾 

```ruby
def recent_three
    posts.order(created_at: :desc).limit(3)
  end
```

- [ ] Also, please include validation test as shown below: example 👇🏾 

```ruby
describe 'validations' do
  it { should validate_presence_of(:name) }
end
```

File: spec/models/post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  it 'The comments_counter should be greater than or equal to zero' do
+    subject.comments_counter = -8
+    expect(subject).to_not be_valid
+  end
+
+  it ' Should test comments_counter to be >= 0' do
+    expect(subject.comments_counter).to be >= 0
+  end
```
Comment (bobb-Rob) [inline spec/models/post_spec.rb:29]: - [ ] Please include unit test for the this method in Post model 👇🏾 

```ruby
def recent_five
    comments.order(created_at: :desc).limit(5)
  end
```

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @RashidMenhas , 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by brytebee — APPROVED** [review]
Hi, @RashidMenhas 👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great UI design.✔️
- App functions as it should.✔️

##

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



##

###  **Optional Suggestions:**
- N/A

##

Good luck! 🎉 

------

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**